modules/core/src/matrix_wrap.cpp

```diff
@@ -1656,6 +1656,14 @@ void _OutputArray::create(int d, const int* sizes, int mtype, int i,
     CV_Error(Error::StsNotImplemented, "Unknown/unsupported array type");
 }
 
+Mat _OutputArray::reinterpretType(int mtype) const
+{
+    mtype = CV_MAT_TYPE(mtype);
+    Mat& m = *(Mat*)obj;
```

Comment:
don't silently assume that `obj` is `Mat*`. Do
`return getMat().reinterpretType(mtype);` instead

Reply:
@vpisarev I had done